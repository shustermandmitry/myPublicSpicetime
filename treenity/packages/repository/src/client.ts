export * from './client/access-token';
export * from './client/create-feathers-client';
export * from './client/simple-client';
export * from './utils/cookies';
export * from './client/socket-custom-methods';
export * from './mods/client';
export * from './client/parse-jwt';
export * from './utils/email-parse';
export { makeServiceActions } from '@/mods/entity/service-entity-actions';
