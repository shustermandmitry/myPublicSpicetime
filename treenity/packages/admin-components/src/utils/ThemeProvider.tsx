/*
 * Copyright (c) 2024. Treenity Inc.
 */

import { setCookie, useCookie } from '@treenity/repository/client';
import { TreeNode, useTree } from '@treenity/tree-api';
import { useStorage } from '@treenity/ui-kit/hooks';
import { Theme, ThemeEntity, ThemeProvider } from '@treenity/ui-kit/theme';
import React, { FC, PropsWithChildren } from 'react';
import useSWR from 'swr';

const defaultThemeName = 'light';

function useLoadTheme(name: string): Theme | undefined {
  const treeApi = useTree();
  const { data: themeNode } = treeApi.useNode(`/sys/themes/${name}`);
  const { data } = useSWR(`${name}-theme` + (themeNode as any)?.$id, () =>
    (themeNode as TreeNode)?.get(ThemeEntity),
  );
  return data;
}

const ThemeProviderLoader: FC<PropsWithChildren> = ({ children }) => {
  const lightTheme = useLoadTheme('light');
  const darkTheme = useLoadTheme('dark');

  const cookies = useCookie();
  const [storedThemeName, _setThemeName] = useStorage<string>(
    'theme',
    cookies.theme || defaultThemeName,
  );
  const setThemeName = (theme: string) => {
    setCookie('theme', theme, 365 * 24 * 60 * 60);
    _setThemeName(theme);
  };

  if (!lightTheme || !darkTheme) return null;

  return (
    <ThemeProvider
      light={lightTheme}
      dark={darkTheme}
      children={children}
      storedThemeName={storedThemeName}
      setThemeName={setThemeName}
    />
  );
};

export default ThemeProviderLoader;
