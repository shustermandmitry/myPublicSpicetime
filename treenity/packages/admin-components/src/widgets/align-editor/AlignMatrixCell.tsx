/*
 * Copyright (c) 2024. Treenity Inc.
 */

import Icon from '@/components/icon';
import useHover from '@/hooks/useHover';
import { css } from '@emotion/react';
import styled from '@emotion/styled';
import { omitProps } from '@treenity/ui-kit/utils';
import { FC } from 'react';
import type {
  AlignCellsType,
  AlignDisplayType,
  AlignOrientationProps,
  IAlignIconProps,
  IAlignMatrixCell,
} from './types';

const row: AlignDisplayType = {
  center_start: 'top-row_outlined',
  left_start: 'top-row_outlined',
  right_start: 'top-row_outlined',
  left_center: 'center-row_outlined',
  center_center: 'center-row_outlined',
  right_center: 'center-row_outlined',
  center_end: 'bottom-row_outlined',
  left_end: 'bottom-row_outlined',
  right_end: 'bottom-row_outlined',
  'space-between_start': 'row-top-space-between_outlined',
  'space-between_center': 'row-center-space-between_outlined',
  'space-between_end': 'row-bottom-space-between_outlined',
  'space-between_stretch': 'space-between_outlined',
  'space-around_start': 'row-top-space-around_outlined',
  'space-around_center': 'row-center-space-around_outlined',
  'space-around_end': 'row-bottom-space-around_outlined',
  'space-around_stretch': 'space-around_outlined',
  center_stretch: 'stretch-horizontal_outlined',
  left_stretch: 'stretch-horizontal_outlined',
  right_stretch: 'stretch-horizontal_outlined',
};

const GeneralAlignIcons: AlignDisplayType = {
  left_start: 'left-column_outlined',
  center_center: 'center-column_outlined',
  right_end: 'right-column_outlined',
  'space-between_start': 'column-bottom-space-between_outlined',
  'space-between_center': 'column-center-space-between_outlined',
  'space-between_end': 'column-top-space-between_outlined',
  'space-between_stretch': 'space-between-vertical_outlined',
  'space-around_start': 'column-bottom-space-around_outlined',
  'space-around_center': 'column-center-space-around_outlined',
  'space-around_end': 'column-top-space-around_outlined',
  'space-around_stretch': 'space-around-vertical_outlined',
  center_stretch: 'stretch-vertical_outlined',
  left_stretch: 'stretch-vertical_outlined',
  right_stretch: 'stretch-vertical_outlined',
};

export const alignIcons: IAlignIconProps = {
  grid: {
    column: {
      left_center: 'left-column_outlined',
      left_end: 'left-column_outlined',
      center_start: 'center-column_outlined',
      center_end: 'center-column_outlined',
      right_start: 'right-column_outlined',
      right_center: 'right-column_outlined',
      ...GeneralAlignIcons,
    },
    row,
  },
  flex: {
    column: {
      left_center: 'center-column_outlined',
      left_end: 'right-column_outlined',
      center_start: 'left-column_outlined',
      center_end: 'right-column_outlined',
      right_start: 'left-column_outlined',
      right_center: 'center-column_outlined',
      ...GeneralAlignIcons,
    },
    row,
  },
};

const MatrixPointIcon = <Icon name="matrix-point_outlined" color="gray700" />;

const AlignMatrixCell: FC<IAlignMatrixCell> = ({
  value,
  orientation,
  selected,
  cellsType,
  onClick,
  display,
}) => {
  const [refCell, hovered] = useHover();
  const iconName = alignIcons[display as keyof IAlignIconProps][orientation][value];

  return (
    <Cell
      ref={refCell}
      onClick={onClick}
      cellsType={cellsType}
      orientation={orientation}
      about={value}
    >
      {iconName &&
        (hovered || selected ? (
          <IconStyled
            name={iconName}
            color="primary"
            cellsType={cellsType}
            orientation={orientation}
          />
        ) : cellsType === 'full' ? (
          MatrixPointIcon
        ) : (
          cellsType === 'triple' && [MatrixPointIcon, MatrixPointIcon, MatrixPointIcon]
        ))}
    </Cell>
  );
};

const IconStyled = styled(Icon, omitProps('cellsType', 'orientation'))<{
  cellsType: AlignCellsType;
  orientation: AlignOrientationProps;
}>`

  && i {
    font-size: ${p =>
      (p.orientation === 'column' && p.cellsType === 'triple') || p.cellsType === 'single'
        ? 54
        : 18}px
`;

const Cell = styled('div', omitProps('cellsType', 'orientation'))<{
  cellsType: AlignCellsType;
  orientation: AlignOrientationProps;
}>`
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  color: ${p => p.theme.colorBorder};

  ${p =>
    p.cellsType === 'triple' &&
    css`
      justify-content: space-around;
    `}

  ${p =>
    p.orientation === 'column' &&
    css`
      flex-direction: column;
    `};

  :where(i) {
    color: ${p => p.theme.colorPrimary};
  }
`;

export default AlignMatrixCell;
