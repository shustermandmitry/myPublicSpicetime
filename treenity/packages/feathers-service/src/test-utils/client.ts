import {
  Application,
  defaultEventMap,
  defaultServiceMethods,
  RealTimeConnection,
  TransportConnection,
} from '@feathersjs/feathers';
import { Service, SocketService } from '@feathersjs/transport-commons/client';
import { runMethod } from '@feathersjs/transport-commons/lib/socket/utils';

export { SocketService };

export default function testClient<Services = any>(options?: any) {
  const defaultService = function (this: Application<any>, name: string) {
    const self = this;
    const events = Object.values(defaultEventMap);
    const connection = {
      emit(method: string, ...args: any[]) {
        return runMethod(self, {} as RealTimeConnection, name, method, args);
      },
    };

    const settings = Object.assign({}, options, {
      events,
      name,
      connection,
      method: 'emit',
    });

    return new Service(settings) as any;
  };

  const initialize = function (app: Application<Services>) {
    app.defaultService = defaultService;
    app.mixins.unshift((service, _location, options) => {
      if (options && options.methods && service instanceof Service) {
        const customMethods = options.methods.filter(name => !defaultServiceMethods.includes(name));

        service.methods(...customMethods);
      }
    });
  };

  initialize.Service = Service;
  initialize.service = defaultService;

  return initialize as TransportConnection<Services>;
}

if (typeof module !== 'undefined') {
  module.exports = Object.assign(testClient, module.exports);
}
