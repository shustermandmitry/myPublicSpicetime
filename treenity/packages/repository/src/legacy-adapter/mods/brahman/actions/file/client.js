import './TelegramActionFileSend.client';
import './TelegramActionFileUpload.client';
import './TelegramFile.client';
