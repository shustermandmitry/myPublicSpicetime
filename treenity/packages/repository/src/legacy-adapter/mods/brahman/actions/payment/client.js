import './TelegramActionGetWallet.client';
import './TelegramActionSendCoins.client';
import './TelegramActionSplitCoins.client';
