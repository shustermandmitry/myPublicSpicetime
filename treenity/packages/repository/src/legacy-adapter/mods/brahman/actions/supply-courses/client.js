import './TelegramActionSupplyCourses.client';
import './TelegramCoursesOperator.client';
import './TelegramActionBecomeOperator.client';
