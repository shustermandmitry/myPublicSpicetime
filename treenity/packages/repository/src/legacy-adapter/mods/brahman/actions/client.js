import './addActionMenu';
import './back/TelegramActionBack.client';
import './broadcast/TelegramActionBroadcast.client';

import './call-command-by-param/TGActionCallCommandByParam.client';
import './crypto/TelegramActionGetRates.client';
// import './voice/video/SynhesizeAudioVideo';
import './emit-text/EmitText.client';
import './file/TelegramActionFileSend.client';
import './file/TelegramActionFileUpload.client';
import './file/TelegramFile.client';
import './forward/TelegramActionForward.client';
import './game/CurrentGame.client';
import './game/Game.client';
import './game/TelegramActionGame.client';
import './get-value/GetValue.client';
import './i18n/TGActionSelectLanguage.client';
import './if-else/TelegramActionIfElse.client';

import './keyword-select/TelegramActionKeywordSelect.client';
import './menu/TelegramActionMenu.client';
import './message/TelegramActionMessage.client';
import './on-error/TelegramActionOnError.client';
import './page/TelegramActionPage.client';
// import './pantaro/client';
import './params/TelegramActionParams.client';
import './payment/client';
import './question/TelegramActionQuestion.client';
import './removeMessage/TelegramActionRemove.client';
import './reset-history/ResetSession.client';
import './reset-session/ResetSession.client';
import './set-telegram-user/TGActionSetTelegramUser.client';
import './set-value/SetValue.client';

import './supply-courses/client';

import './tag/SetTag';
import './TelegramAction.client';
import './timers/TgActionSetTimer';

import './timers/timers.service.client';

import './translate/TString.client';
import './voice/tinkoff/TinkoffVoiceRecognition.client';
import './voice/tinkoff/TinkoffVoiceSynthesize.client';

import './voice/yandex/YandexVoiceRecognition.client';
import './voice/yandex/YandexVoiceSynthesize.client';

import './ym-statistics/TelegramPageStatistics.client';
