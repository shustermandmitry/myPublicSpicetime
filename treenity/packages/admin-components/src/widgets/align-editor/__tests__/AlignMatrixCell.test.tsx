/*
 * Copyright (c) 2024. Treenity Inc.
 */

import ThemeProvider from '@/utils/ThemeProvider';
import { jest } from '@jest/globals';
import { fireEvent, render } from '@testing-library/react';
import { CookieContext, createCookies, parseCookies, setCookie } from '@treenity/repository/client';
import AlignMatrixCell, { alignIcons } from '../AlignMatrixCell';
import type { AlignOrientationProps, AlignVariation } from '../types';

const cookies = createCookies(parseCookies(document.cookie), setCookie);

const defaultValue: AlignVariation = 'left_start';

let valueMock: AlignVariation = defaultValue;

const updateMock = jest.fn((value: AlignVariation) => {
  valueMock = value;
  return value;
});

const orientation: AlignOrientationProps[] = ['row', 'column'];
const value = 'left_start';

describe('AlignMatrixCell', () => {
  beforeEach(() => {
    updateMock.mockClear();
    valueMock = defaultValue;
  });

  it('render AlignMatrixCell component', async () => {
    const { container } = render(
      <CookieContext.Provider value={cookies}>
        <ThemeProvider>
          <AlignMatrixCell
            display={'flex'}
            key={0}
            value={value}
            orientation={orientation[0]}
            selected={false}
            cellsType="full"
            onClick={() => updateMock(value)}
          />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

    const root = container.querySelector(`body > div > div`);
    const icon = root!.querySelector(`i`);

    expect(root).toMatchSnapshot();
    expect(root).toBeInTheDocument();
    expect(icon).toBeInTheDocument();
    expect(icon).toHaveClass('icon-matrix-point_outlined');
    expect(icon).not.toHaveClass(`icon-${alignIcons['flex'][orientation[0]][value]}`);
  });

  it('change selected', async () => {
    const { container } = render(
      <CookieContext.Provider value={cookies}>
        <ThemeProvider>
          <AlignMatrixCell
            display={'flex'}
            key={0}
            value={value}
            orientation={orientation[0]}
            selected={true}
            cellsType="full"
            onClick={() => updateMock(value)}
          />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

    const root = container.querySelector(`body > div > div`);
    const icon = root!.querySelector(`i`);

    expect(root).toMatchSnapshot();
    expect(root).toBeInTheDocument();
    expect(icon).toBeInTheDocument();
    expect(icon).not.toHaveClass('icon-matrix-point_outlined');
    expect(icon).toHaveClass(`icon-${alignIcons['flex'][orientation[0]][value]}`);
  });

  it('change selected and orientation', async () => {
    const { container } = render(
      <CookieContext.Provider value={cookies}>
        <ThemeProvider>
          <AlignMatrixCell
            display={'flex'}
            key={0}
            value={value}
            orientation={orientation[1]}
            selected={true}
            cellsType="full"
            onClick={() => updateMock(value)}
          />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

    const root = container.querySelector(`body > div > div`);
    const icon = root!.querySelector(`i`);

    expect(root).toMatchSnapshot();
    expect(root).toBeInTheDocument();
    expect(icon).toBeInTheDocument();
    expect(icon).not.toHaveClass('icon-matrix-point_outlined');
    expect(icon).toHaveClass(`icon-${alignIcons['flex'][orientation[1]][value]}`);
  });

  it('change value', async () => {
    const newValue = 'center_start';

    const { container } = render(
      <CookieContext.Provider value={cookies}>
        <ThemeProvider>
          <AlignMatrixCell
            display={'flex'}
            key={0}
            value={newValue}
            orientation={orientation[0]}
            selected={true}
            cellsType="full"
            onClick={() => updateMock(value)}
          />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

    const root = container.querySelector(`body > div > div`);
    const icon = root!.querySelector(`i`);

    expect(root).toMatchSnapshot();
    expect(root).toBeInTheDocument();
    expect(icon).toBeInTheDocument();
    expect(icon).not.toHaveClass('icon-matrix-point_outlined');
    expect(icon).toHaveClass(`icon-${alignIcons['flex'][orientation[0]][newValue]}`);
  });

  it('hovered', async () => {
    const newValue = 'center_start';

    const { container } = render(
      <CookieContext.Provider value={cookies}>
        <ThemeProvider>
          <AlignMatrixCell
            display={'flex'}
            key={0}
            value={newValue}
            orientation={orientation[0]}
            selected={valueMock === newValue}
            cellsType="full"
            onClick={() => updateMock(value)}
          />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

    const root = container.querySelector(`body > div > div`);
    const icon = root!.querySelector(`i`);

    expect(root).toMatchSnapshot();
    expect(root).toBeInTheDocument();
    expect(icon).toBeInTheDocument();
    expect(icon).toHaveClass('icon-matrix-point_outlined');
    expect(icon).not.toHaveClass(`icon-${alignIcons['flex'][orientation[0]][newValue]}`);

    fireEvent.mouseEnter(root!);

    const icon2 = root!.querySelector(`i`);
    expect(icon2).toBeInTheDocument();
    expect(icon2).not.toHaveClass('icon-matrix-point_outlined');
    expect(icon2).toHaveClass(`icon-${alignIcons['flex'][orientation[0]][newValue]}`);
  });

  it('clicked', async () => {
    const newValue = 'center_start';

    const { container, rerender } = render(
      <CookieContext.Provider value={cookies}>
        <ThemeProvider>
          <AlignMatrixCell
            display={'flex'}
            key={0}
            value={newValue}
            orientation={orientation[0]}
            selected={valueMock === newValue}
            cellsType="full"
            onClick={() => updateMock(value)}
          />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

    const root = container.querySelector(`body > div > div`);
    const icon = root!.querySelector(`i`);

    expect(root).toMatchSnapshot();
    expect(root).toBeInTheDocument();
    expect(icon).toBeInTheDocument();
    expect(icon).toHaveClass('icon-matrix-point_outlined');
    expect(icon).not.toHaveClass(`icon-${alignIcons['flex'][orientation[0]][newValue]}`);

    fireEvent.click(root!);

    rerender(
      <CookieContext.Provider value={cookies}>
        <ThemeProvider>
          <AlignMatrixCell
            display={'flex'}
            key={0}
            value={'left_start'}
            orientation={orientation[0]}
            selected={valueMock === 'left_start'}
            cellsType="full"
            onClick={() => updateMock(value)}
          />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

    expect(updateMock.mock.calls).toHaveLength(1);

    expect(updateMock.mock.calls.at(-1)).toEqual(['left_start']);

    const icon2 = root!.querySelector(`i`);
    expect(icon2).toBeInTheDocument();
    expect(icon2).not.toHaveAttribute('icon-matrix-point_outlined');
    expect(icon2).toHaveClass(`icon-${alignIcons['flex'][orientation[0]]['left_start']}`);
  });
});
