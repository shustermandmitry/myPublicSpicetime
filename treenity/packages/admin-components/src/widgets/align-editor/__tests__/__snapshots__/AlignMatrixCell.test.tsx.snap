// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlignMatrixCell change selected 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  cursor: pointer;
  color: #d9d9d9;
}

.emotion-0:where(i) {
  color: #1677ff;
}

.emotion-1 [class^='icon-'],
.emotion-1 [class*=' icon-'] {
  font-size: 15px;
}

.emotion-1.emotion-1 i {
  font-size: 18px;
}

.emotion-2 {
  -webkit-transition: -webkit-transform 0.2s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  transition: transform 0.2s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  color: #1677ff;
}

<div
  about="left_start"
  class="emotion-0"
>
  <span
    class="anticon emotion-1"
  >
    <i
      class="icon-top-row_outlined emotion-2"
    />
  </span>
</div>
`;

exports[`AlignMatrixCell change selected and orientation 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  cursor: pointer;
  color: #d9d9d9;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-0:where(i) {
  color: #1677ff;
}

.emotion-1 [class^='icon-'],
.emotion-1 [class*=' icon-'] {
  font-size: 15px;
}

.emotion-1.emotion-1 i {
  font-size: 18px;
}

.emotion-2 {
  -webkit-transition: -webkit-transform 0.2s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  transition: transform 0.2s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  color: #1677ff;
}

<div
  about="left_start"
  class="emotion-0"
>
  <span
    class="anticon emotion-1"
  >
    <i
      class="icon-left-column_outlined emotion-2"
    />
  </span>
</div>
`;

exports[`AlignMatrixCell change value 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  cursor: pointer;
  color: #d9d9d9;
}

.emotion-0:where(i) {
  color: #1677ff;
}

.emotion-1 [class^='icon-'],
.emotion-1 [class*=' icon-'] {
  font-size: 15px;
}

.emotion-1.emotion-1 i {
  font-size: 18px;
}

.emotion-2 {
  -webkit-transition: -webkit-transform 0.2s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  transition: transform 0.2s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  color: #1677ff;
}

<div
  about="center_start"
  class="emotion-0"
>
  <span
    class="anticon emotion-1"
  >
    <i
      class="icon-top-row_outlined emotion-2"
    />
  </span>
</div>
`;

exports[`AlignMatrixCell clicked 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  cursor: pointer;
  color: #d9d9d9;
}

.emotion-0:where(i) {
  color: #1677ff;
}

.emotion-1 [class^='icon-'],
.emotion-1 [class*=' icon-'] {
  font-size: 15px;
}

.emotion-2 {
  -webkit-transition: -webkit-transform 0.2s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  transition: transform 0.2s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  color: gray700;
}

<div
  about="center_start"
  class="emotion-0"
>
  <span
    class="anticon undefined emotion-1"
  >
    <i
      class="icon-matrix-point_outlined emotion-2"
    />
  </span>
</div>
`;

exports[`AlignMatrixCell hovered 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  cursor: pointer;
  color: #d9d9d9;
}

.emotion-0:where(i) {
  color: #1677ff;
}

.emotion-1 [class^='icon-'],
.emotion-1 [class*=' icon-'] {
  font-size: 15px;
}

.emotion-2 {
  -webkit-transition: -webkit-transform 0.2s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  transition: transform 0.2s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  color: gray700;
}

<div
  about="center_start"
  class="emotion-0"
>
  <span
    class="anticon undefined emotion-1"
  >
    <i
      class="icon-matrix-point_outlined emotion-2"
    />
  </span>
</div>
`;

exports[`AlignMatrixCell render AlignMatrixCell component 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  cursor: pointer;
  color: #d9d9d9;
}

.emotion-0:where(i) {
  color: #1677ff;
}

.emotion-1 [class^='icon-'],
.emotion-1 [class*=' icon-'] {
  font-size: 15px;
}

.emotion-2 {
  -webkit-transition: -webkit-transform 0.2s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  transition: transform 0.2s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  color: gray700;
}

<div
  about="left_start"
  class="emotion-0"
>
  <span
    class="anticon undefined emotion-1"
  >
    <i
      class="icon-matrix-point_outlined emotion-2"
    />
  </span>
</div>
`;
